app/services/cert/rhsm_client.rb

```diff
@@ -10,7 +10,7 @@ def initialize(cert)
     end
 
     def uuid
-      drop_cn_prefix_from_subject(@cert.subject.to_s)
+      @uuid ||= @cert.subject.to_a.detect { |name, _, _| name == 'CN' }&.second
```

Comment:
It would be nice if we could align this with [Foreman's CertificateExtract class](https://github.com/theforeman/foreman/blob/develop/app/services/certificate_extract.rb). It looks like we could actually just have a single class between both projects.